### PR TITLE
workflow: add clippy and cargo-deny

### LINKED ISSUE #12: Umbrella issus for basic function
## Description
This is an umbrella issue to track some basic functions.

## CI/CD
- [x] Run `Cargo clippy -D warning` and `Cargo deny` in Github action  #22 @dierbei 
- [ ] Prepare for the physical machine: Postpone
- [x] #4 @flyflypeng 

## Functions
- [x] Privileged container in vmm-sandboxer : Supported
- [x] Resource usage stats for both container and sandbox : Supported
- [x] Time synchronization between Host and Guest VM @Burning1020 #50 
- [x] Limit VM process into cgroup filesystem @flyflypeng #51 
- [x] Sandbox files persist and recover @Vanient #40

### PR DESCRIPTION
**Pull request by dierbei**
adds a GitHub workflow that performs clippy linting and cargo-deny auditing on Rust code.

Fixes #12

### CODE REVIEW

**Comment by Burning1020** [comment]
@dierbei Thanks for PR. Could you move this code to .github/workflows/ci.yml ? We already have some checks there.

**Comment by dierbei** [comment]
No problem, moved to ci.yml.

**Comment by dierbei** [comment]
There is a problem, need to operate under the subfolder.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,12 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+      - name: Linting
+        run: cargo clippy -- -D warnings
```
Comment (Burning1020) [inline .github/workflows/ci.yml:24]: Please add `-- -D warnings` to L43~L47

Comment (dierbei) [reply .github/workflows/ci.yml]: I added some deny.toml files, modified Clippy.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,11 +47,11 @@ jobs:
           pushd quark && cargo +nightly fmt --all -- --check --files-with-diff && popd
       - name: Clippy
         run: |
-          pushd vmm/sandbox && cargo clippy --all-features && popd
-          pushd vmm/task && cargo clippy --all-features && popd
-          pushd shim && cargo clippy --all-features && popd
-          pushd wasm && cargo clippy --all-features && popd
-          pushd quark && cargo clippy --all-features && popd
+          pushd vmm/sandbox && cargo clippy -- -D warnings && popd
```
Comment (Burning1020) [inline .github/workflows/ci.yml:50]: Still need `--all-features`.
LGTM on others.

Comment (dierbei) [reply .github/workflows/ci.yml]: It has been modified.

**Comment by Burning1020** [comment]
@dierbei Please check username and email in your git config and rebase commits into one.

**Comment by dierbei** [comment]
I refer to: `https://rust-coding-guidelines.github.io/rust-coding-guidelines-zh/safe-guides/Appendix/templates/deny.toml.html`

**Comment by dierbei** [comment]
update deny.toml content, look again

**Pull request closed without merging**